### ISSUE TITLE
Split the topology-defaults file

### ISSUE THREAD

**Issue by ipspace** [ipspace/netlab#701]
Topology-defaults file is becoming way too large (and will get much bigger when we implement #700). We should split it into numerous smaller files and combine them either at package publishing time or at runtime (depending on how time-consuming the whole thing is).

Implementation is "trivial":

* Topology defaults can have **include** attribute anywhere
* **include** attribute specifies a glob relative to the topology-defaults.yml file
* All files matched by the glob are read and inserted as a single key-value pair into the defaults at the level of the **include** directive. The key is the name of the file (minus the file type), the value is whatever was read from the file.

For example:

```
include: modules/*.yml
devices:
  include: devices/*.yml
providers:
  include: providers/*.yml
```

Caveats:
* This process would work only for package topology-defaults and would not be documented outside of developer documentation
* Included files cannot have further **include** attributes. We have to stop somewhere, plus this nicely stops infinite recursion ;)

Packaging optimization:
* If needed, we'd have a script that would generate fully-evaluated topology-defaults.yml (or even a JSON file -- JSON is much faster to read) to be included in published Python packages.

**Comment by ssasso** [comment]
Can't we simply "reuse" this? (and use it also for standard topology files?)
https://pypi.org/project/pyyaml-include/

**Comment by ipspace** [comment]
> Can't we simply "reuse" this? https://pypi.org/project/pyyaml-include/

And this is why I always open an issue before I start writing code ;) -- I'm always missing something obvious.

Unfortunately we can't use that package:

* It uses YAML directive to include files, so we cannot include stuff at the top level (which is needed for modules)
* It includes multiple files as a list whereas we need them as a dictionary.

As for "and use it also for standard topology files?" I have no problem with that. Will do.

**Comment by ipspace** [comment]
Working on in in the 'include-701' branch.

**Comment by ipspace** [comment]
Implemented in #719
